### PR TITLE
Adds Prisma, Updates data modal and implements caching  

### PR DESCRIPTION
**Pull request by Royal-lobster**
Closes https://github.com/EveripediaNetwork/issues/issues/1237
Closes https://github.com/EveripediaNetwork/issues/issues/1243
Closes https://github.com/EveripediaNetwork/issues/issues/1235
Closes https://github.com/EveripediaNetwork/issues/issues/1165

# Changes
- adds prisma to project
- migrates from supabase client for prisma 
- updates database model
- adapts data changes across the app

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+model langchain_split {
+  id        BigInt                 @id @default(autoincrement())
+  wikiid    String?
+  title     String?
+  content   String?
+  embedding Unsupported("vector")?
+}
+
+model wiki_embedding {
```
Comment (kesar) [inline prisma/schema.prisma:26]: not sure if part of this PR but I suggested several changes in db structure that should be applied. We can do a follow up PR, but db will change a lot. You can sync w @Yadheedhya06 about our discussion

Comment (Royal-lobster) [reply prisma/schema.prisma:26]: Cool !

**Review by kesar — APPROVED** [review]

**Comment by Softdev1** [comment]
why is this PR still open @Royal-lobster

**Comment by Royal-lobster** [comment]
Database and functions are now done, everything is working as expected. merging

**Pull request merged**